docs/sharding.md

```diff
@@ -0,0 +1,107 @@
+
+    ```json
+    {
+        "AzureSignalRServiceTransportType":"Persistent"
+    }
+    ```
+
+2. Configure multiple SignalR Service endpoints entries in your configuration.
+
+    We use a [`ServiceEndpoint`](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs) object to represent a SignalR Service instance. You can define an service endpoint with its `<Name>` and `<EndpointType>` in the entry key, and the connection string in the entry value. The keys are in the following format : 
+
+    ```
+    Azure:SignalR:Endpoints:<Name>:<EndpointType>
+    ```
+
+    `<EndpointType>` is optional and is `primary` by default. See samples below:
+        
+    ```json
+    {
+        "Azure:SignalR:Endpoints:EastUs":"<ConnectionString>",
+        
+        "Azure:SignalR:Endpoints:EastUs2:Secondary":"<ConnectionString>",
+        
+        "Azure:SignalR:Endpoints:WestUs:Primary":"<ConnectionString>"
+    }
+    ```
+
+    > When you configure Azure SignalR endpoints in the App Service on Azure portal, don't forget to replace `":"` with `"__"`, the double underscore in the keys. For reasons, see [Environment variables](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0#environment-variables).
+
+## Route algorithm
+
+### Default behavior
+By default, the SDK uses the [DefaultEndpointRouter](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR/EndpointRouters/DefaultEndpointRouter.cs) to pick up endpoints.
+
+* Client routing: Randomly select one endpoint from **primary online** endpoints. If all the primary endpoints are offline, then randomly select one **secondary online** endpoint. If the selection fails again, then exception is thrown.
+
+* Server message routing: All service endpoints are returned.
+
+### Customization
+We support cutomization of route algoritm in C# language. 
+
+Here are the steps:
+* Implement a cutomized router. You can leverage information provided from [`ServiceEndpoint`](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs) to make routing decision. See guide here: [customize-route-algorithm](https://github.com/Azure/azure-signalr/blob/dev/docs/sharding.md#customize-route-algorithm).
+
+* Register the router to DI container.
+```cs
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Azure.SignalR;
+using Microsoft.Extensions.DependencyInjection;
+
+[assembly: FunctionsStartup(typeof(SimpleChatV3.Startup))]
+namespace SimpleChatV3
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.Services.AddSingleton<IEndpointRouter, CustomizedRouter>();
```

Comment:
add comment to make cx focus on this line
// Register the router to DI container